src/controllers/authenticationController.ts

```diff
@@ -200,6 +207,90 @@ export default class AuthenticationController {
       sessionTrackingId
     );
 
+    const errorOrMaybeAssertionRef = this.lollipopParams.isLollipopEnabled
+      ? await this.sessionStorage.getLollipopAssertionRefForUser(user)
+      : E.right(O.none);
+
+    if (E.isLeft(errorOrMaybeAssertionRef)) {
+      return ResponseErrorInternal(
+        "Error retrieving previous lollipop configuration"
+      );
+    }
+
+    if (
+      this.lollipopParams.isLollipopEnabled &&
+      O.isSome(errorOrMaybeAssertionRef.right)
+    ) {
+      // Sending a revoke message for previous assertionRef related to the same fiscalCode
+      // This operation is fire and forget
+      this.lollipopParams.lollipopService
+        .revokePreviousAssertionRef(errorOrMaybeAssertionRef.right.value)
+        .catch(err => {
+          log.error(
+            "acs: error sending revoke message for previous assertionRef [%s]",
+            err
+          );
+        });
+    }
+
+    // Delete the reference to CF and assertionRef for lollipop.
+    // This operation must be performed even if the lollipop FF is disabled
+    // to avoid inconsistency on CF-key relation if the FF will be re-enabled.
+    const errorOrDelLollipop = await this.sessionStorage.delLollipopAssertionRefForUser(
+      spidUser.fiscalNumber
+    );
+
+    if (E.isLeft(errorOrDelLollipop) || !errorOrDelLollipop.right) {
+      return ResponseErrorInternal("Error on LolliPoP initialization");
+    }
+
+    const maybeRequestId = getRequestIDFromResponse(
+      new DOMParser().parseFromString(
+        errorOrSpidUser.right.getAssertionXml(),
+        "text/xml"
+      )
+    );
```

Comment:
Since Xml is parsed to create SpidUser, could be useful to add a property that maps `InResponseTo` attribute?

Reply:
The creation of `SpidUser` object is hided inside the strategy, maybe in the not extended/overrided part of `io-spid-commons`. The custom hooks are in prevalidation fase, the SpidUser is builded after. We can take a look